### PR TITLE
create endpoints

### PR DESCRIPTION
**Pull request by jflora91**

### CODE REVIEW

**Comment by tsbatista** [comment]
There is a little detail that is missing, the multi tenancy aspect of the service. We need most of our paths scoped to a single tenant - with the exception of the ones that are meant for the management of base configurations

File: README.md [hunk 1]
```diff
@@ -1,2 +1,30 @@
 # config-server
 Hierarchical json configuration management for multitenant environments
+
+---
+
+### URI endpoints
+
+|paths|||
+|:-|:-|:-|
+|config/|GET, POST|Get all versions, or create a new one|
```
Comment (ncarapeto) [inline README.md]: I really don't see a need for a GET that retrieves all versions! 
On 99% of request I don't care about the versions, I just want the latest. Plus I should not be forced to know which one it is.

I'd rather assume this endpoint is a GET over the latest and a POST of the latest

File: README.md [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+|config/?version={version}|GET|Get a specific version|
+|config/{id}/scopes/?scope={scope}|GET|Get a specific scope|
+|bases/|GET, POST|Get all versions of base json file, or create a new version of base|
```
Comment (ncarapeto) [inline README.md]: Are there multiple bases? Shouldn't it be just 1 with multiple versions? If so why: bases/{id}
Plus same as above, don't think there is a need to retrieve a list of all the versions, at this endpoint, just the latest

**Review by ncarapeto — COMMENTED** [review]
Please review

File: README.md (hunk 1 above)
Comment (jflora91) [reply README.md]: It's a good point, I agree with you.
Lets save this topic for tomorrow meeting/call, there is something else I would like to talk about too

File: README.md (hunk 2 above)
Comment (jflora91) [reply README.md]: i see this "bases" has default/first versions and from them we can start making versions

Comment (ncarapeto) [reply README.md]: i'm asking about the {id}

File: README.md (hunk 1 above)
Comment (daveseco7) [reply README.md]: I also agree with this, most of the times i just want to get a configuration (meaning that i want to retrieve the latest version)

**Review by ncarapeto — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -24,3 +24,5 @@ hs_err_pid*
 
 # IntelliJ IDEA
 .idea/
+
```
Comment (ncarapeto) [inline .gitignore]: This file seems to be extra on this PR

File: README.md [hunk 4]
```diff
@@ -1,2 +1,28 @@
...
+
+---
+
+### URI endpoints
+
+|paths|||
+|:-|:-|:-|
+|config/|GET, POST|Get last versions, or create a new one|
+|config/?version={version}|GET|Get a specific version|
+|config/scopes/?scope={scope}|GET|Get a specific scope|
```
Comment (ncarapeto) [inline README.md]: `config/scopes/?scope={scope}`
This does not make a lot of sense to me... it should be either a path param 
`config/scopes/{scope}`
or a query param
`config/?scope={scope}`
not a mix of both

Personally I prefer the path param on this case.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,28 @@
 # config-server
 Hierarchical json configuration management for multitenant environments
+
+---
+
+### URI endpoints
+
+|paths|||
+|:-|:-|:-|
+|config/|GET, POST|Get last versions, or create a new one|
```
Comment (ncarapeto) [inline README.md]: typo: version not versions
plus you might want to add that it retrieves all the scopes of that version

File: README.md [hunk 6]
```diff
@@ -1,2 +1,28 @@
...
+|config/scopes/?scope={scope}|GET|Get a specific scope|
+|bases/|GET|Get base json file|
+
+##### Example of scenes:
+
+* update scopes
+    * make a post with some update scopes (POST config/);
+    * check if DTO is valid(*data, base_id, tenant*);
+        * *data*: valid json structure;
+        * *base_id*: exist and coherent with *data* structure;
```
Comment (ncarapeto) [inline README.md]: *base_id*: exist and coherent with *data* structure; What is this?

File: .gitignore (hunk 3 above)
Comment (jflora91) [reply .gitignore]: > This file seems to be extra on this PR

you are right, I'll fix this

File: README.md (hunk 6 above)
Comment (jflora91) [reply README.md]: > _base_id_: exist and coherent with _data_ structure; What is this?

done and new PR fix this issue in data model

File: README.md (hunk 5 above)
Comment (jflora91) [reply README.md]: > typo: version not versions
> plus you might want to add that it retrieves all the scopes of that version

description fixed, check if is ok to be merged, thanks

File: README.md (hunk 2 above)
Comment (jflora91) [reply README.md]: updated just to get a base, check if is ok to be merged, thanks

File: README.md (hunk 4 above)
Comment (jflora91) [reply README.md]: I follow your tip, it's fixed to get a scope by path param, thanks

File: .gitignore (hunk 3 above)
Comment (tsbatista) [reply .gitignore]: no need to do it now, just take care next time!

**Comment by ncarapeto** [comment]
The Model looks good just the extra .gitignore doesn't

**Comment by jflora91** [comment]
> The Model looks good just the extra .gitignore doesn't

solved, check if is ok to merge.

**Review by daveseco7 — APPROVED** [review]

**Pull request merged**